TWNIA2/2. Tutorials/3. 使用 Conda 管理套件與執行 Job.md

```diff
@@ -1 +1,229 @@
+```bash
+# When Conda is no longer used, delete the remaining Conda installation files/configuration files (please use it carefully!! Make sure there is no need for files in this path before using this command)
+# Check the contents of this path file first (usually only ~/.conda)
+$ ls ~/.condarc ~/.conda ~/.continuum
+
+# Then, delete the files
+$ rm -rf ~/.condarc ~/.conda ~/.continuum
+```
+
+### Step 2. Slurm job script with Conda
+
+Once you have prepared the Conda environment and computing program, please follow the steps to write a job script using Slurm Workload Manager to request resources, queue schedules, and submit jobs according to your needs.
+
+
+The `.sh` file is the job script submitted by Slurm, the content is divided into two parts:
+
+1. **Information of Job, project and resources**: job name, Number of nodes, Number of jobs running per node, Number of GPUs per node, longest job running time, project ID, and queue name.
+3. **The content of the job**
+
+
+Write the following content and use the virtual environment `mytf_gpu` created by the above steps.
+
+
+Run the following edit command > Press <kbd>i</kbd> key > Copy and paste the example > Press <kbd>ESC</kbd> key > Enter `:wq!` to save and exit. We're done!
+
+
+```bash
+$ vim <FILE_NAME>.sh
+```
+
+:::info 
+<i class="fa fa-paperclip fa-20" aria-hidden="true"></i> **Note:** You may write `.sh`  files using the editor you're used to, the example uses vim to operate.
+:::
+
+- TensorFlow 2 with 1 node / 8GPUs
+```bash
+#!/bin/bash
+#SBATCH --job-name=Hello_TWCC    ## job name
+#SBATCH --nodes=1                ## request 1 node
+#SBATCH --cpus-per-task=32       ## request 32 CPUs for that task
+#SBATCH --gres=gpu:8             ## request 8 GPUs per node
+#SBATCH --time=00:10:00          ## Run for up to 10 minutes (remember to change it after testing, or delete this line directly)
+#SBATCH --account=PROJECT_ID     ## PROJECT_ID (e.g.,  MST108XXX). Billing will also be based on this project ID
+#SBATCH --partition=gtest        ## gtest is used for testing queue. After subsequent tests, it can be changed to gp1d (maximum running 1 day), gp2d (maximum running 2 days), gp4d (maximum running 4 days)
+
+module purge
+module load miniconda3
+conda activate mytf_gpu
+
+# The programs of most of conda users do not communicate through MPI
+# You do not need to add srun/mpirun, just add the commands that need to be run
+
+python $(your command) 
+```
+
+
+:::info
+<i class="fa fa-paperclip fa-20" aria-hidden="true"></i> **Note:** 
+1. Adding the following content to the script file header, you may add an email to get the job state notification:
+    ```bash
+    #SBATCH --mail-type=ALL
+    #SBATCH --mail-user=$Your_email
+    ```
+2. The ratio of the requested resources will be based on the number of GPUs you assign, and will be allocated based on the ratio of 1 GPU: 4 CPU: 90 GB Memory. For example,
+    > Request 1 GPU, 90 GB Memory will be automatically allocated
```

Comment:
Request 1 GPU, you will get 4 CPU cores and 90 GB Memory automatically allocated.